src/models/User.ts

```diff
@@ -0,0 +1,81 @@
+import firebase, { firestore } from 'firebase';
+import FirestoreModel from './FirestoreModel';
+import {
+  Grade,
+  ScholarshipType,
+  Gender,
+  ScholarshipAmount,
+  State,
+} from './enums';
+
+export enum Role {
+  READER,
+  CONTRIBUTOR,
+  MODERATOR,
+  ADMIN,
+}
+export enum NotificationType {
+  NONE,
+  EMAIL,
+  MOBILE,
+}
+
+interface UserData {
+  DOB?: Date;
+  schoolYear: Grade;
+  role: Role;
+  gender: Gender;
+  school?: string;
+  state?: string;
+  majors?: string[];
+  scholarshipPrefs: ScholarshipPreferences;
+  notificationSettings: NotificationSettings;
+}
+
+interface ScholarshipPreferences {
+  type: ScholarshipType[];
+  major?: string[];
+  amount: {
+    type: ScholarshipAmount;
+    min?: number;
+    max?: number;
+  };
```

Comment:
maybe `ScholarshipAmount` should be this type? And then `ScholarshipAmountType` a separate type